### PR TITLE
feat: Export description as comment for Android

### PR DESCRIPTION
**Pull request by vaclavsauer**
Implementation of https://github.com/tolgee/tolgee-platform/issues/2391

I'm not sure if tests are OK. A lot of them failed because there is a difference in line endings  on windows.

### CODE REVIEW

File: backend/data/src/test/kotlin/io/tolgee/unit/formats/android/out/AndroidXmlFileExporterTest.kt [hunk 1]
```diff
@@ -160,13 +161,13 @@ class AndroidXmlFileExporterTest {
         )
         add(
           languageTag = "cs",
-          keyName = "percent and paceholders",
-          text =
-            "I am not just a percent {name} % sign!",
+          keyName = "percent and placeholders",
+          text = "I am not just a percent {name} % sign!",
+          description = "This is a description",
```
Comment (vaclavsauer) [inline backend/data/src/test/kotlin/io/tolgee/unit/formats/android/out/AndroidXmlFileExporterTest.kt:166]: had to do this this way insted of 
```
{
  key.description = "Description"
}
```
Is it OK?

File: backend/data/src/main/kotlin/io/tolgee/service/export/dataProvider/ExportTranslationView.kt [hunk 2]
```diff
@@ -8,4 +8,5 @@ class ExportTranslationView(
   val state: TranslationState = TranslationState.TRANSLATED,
   val key: ExportKeyView,
   val languageTag: String = "en",
+  val description: String? = null,
```
Comment (vaclavsauer) [inline backend/data/src/main/kotlin/io/tolgee/service/export/dataProvider/ExportTranslationView.kt:11]: Also added description here for the test

File: backend/data/src/test/kotlin/io/tolgee/unit/formats/android/out/AndroidXmlFileExporterTest.kt [hunk 3]
```diff
@@ -23,8 +23,9 @@ class AndroidXmlFileExporterTest {
     |<resources xmlns:xliff="urn:oasis:names:tc:xliff:document:1.2">
     |  <string name="key1">Ahoj! I%d, %s, %e, %f</string>
     |  <string name="percent_no_placeholders">I am just a percent \% sign!</string>
-    |  <string name="percent_and_paceholders">I am not just a percent %s %% sign!</string>
-    |  <string name="percent_and_paceholders_and_tags"><![CDATA[I am not just a percent <b>%s</b> %% sign!]]></string>
+    |  <!--This is a description-->
```
Comment (JanCizmar) [inline backend/data/src/test/kotlin/io/tolgee/unit/formats/android/out/AndroidXmlFileExporterTest.kt:26]: You're testing only the plain string comment, but not plural and string array item.

File: backend/data/src/main/kotlin/io/tolgee/formats/android/out/AndroidStringsXmlFileWriter.kt [hunk 4]
```diff
@@ -25,6 +21,10 @@ class AndroidStringsXmlFileWriter(private val model: AndroidStringsXmlModel) {
   private fun Element.addToElement(unit: Map.Entry<String, AndroidXmlNode>) {
     when (val node = unit.value) {
       is StringUnit -> {
+        val comment = node.comment
+        if (!comment.isNullOrEmpty()) {
```
Comment (JanCizmar) [inline backend/data/src/main/kotlin/io/tolgee/formats/android/out/AndroidStringsXmlFileWriter.kt:25]: It wold be great to Create a private extension function Element.comment, which will add the comment only if it's not null to remove the deduplicate the duplicated null checking logic.

**Review by JanCizmar — CHANGES_REQUESTED** [review]

**Comment by Anty0** [comment]
Superseded by  #2560

**Pull request closed without merging**